src/lib/crawler/crawler.test.ts

```diff
@@ -1,32 +1,133 @@
-import { WarmaneCrawler } from "./crawler";
+import { WarmaneCrawler, MatchSummary, MatchDetails, CharacterDetail } from "./crawler";
 import * as fs from "fs";
 import * as path from "path";
 import axios from "axios";
+import Bottleneck from "bottleneck";
 
 jest.mock("axios");
 const mockedAxios = axios as jest.Mocked<typeof axios>;
-
+//
 describe("crawler tests", () => {
-  test("when an html document is returned with matches, it can return a list of match IDs", async () => {
-    // storing data from test.html file as string
+    test("The crawler can return a list of match summaries (which contain match IDs) when provided an HTML document containing a given player's match information.", async () => {
+      // storing data from test.html file as string
+      const testHtml = fs.readFileSync(
+        path.resolve(__dirname, "test.html"),
+        "utf-8"
+      );
+  
+      mockedAxios.get.mockImplementation(async () => ({
+        data: testHtml,
+      }));
+  
+      const crawler = new WarmaneCrawler();
+      const matchSummaries = await crawler.getMatchSummaries({
+        character: "SomeCharacter",
+        realm: "Blackrock",
+      });
+  
+      expect(matchSummaries.length).toBe(2);
+      expect(matchSummaries.map((summary) => summary.matchId)).toEqual(
+        expect.arrayContaining(["24671286", "25750028"])
+      );
+    });
+
+  test("WarmaneCrawler does not exceed 32 concurrent operations", async () => {
+    // mocking fetchMatchData method to simulate delay
+    const originalFetchMatchData = WarmaneCrawler.prototype.fetchMatchData;
+    WarmaneCrawler.prototype.fetchMatchData = jest.fn(
+      async (matchId: string) => {
+        return new Promise((resolve) =>
+        setTimeout(() => resolve([] as CharacterDetail[]), 100)
+        );
+      }
```

Comment:
This actually accomplishes the same thing that you're doing, but it doesn't introduce 100ms of latency each time it's called.

`async` implicitly returns a `Promise` so you shouldn't need to `new Promise()` here. Also, we're testing the inputs and outputs of functions. We don't need to simulate network latency with `setTimeout()`
```suggestion
async () => []
```